core/coreunix/add.go

```diff
@@ -467,8 +464,8 @@ func (adder *Adder) addFile(file files.File) error {
 	}
 	adder.liveNodes++
 
-	if file.IsDirectory() {
-		return adder.addDir(file)
+	if dir, ok := file.(files.Directory); ok {
```

Comment:
Nit: while a larger edit, changing this function to a type switch would (IMO) make it easier to read.